environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaNetworkProvider.java

```diff
@@ -4,15 +4,24 @@
 
 import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
 import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.InstanceGroupNetworkRequest;
 import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.network.NetworkRequest;
 
 public interface FreeIpaNetworkProvider {
 
-    NetworkRequest provider(EnvironmentDto environment);
+    NetworkRequest provider(EnvironmentDto environment, boolean multiAzRequired);
+
+    default InstanceGroupNetworkRequest providerByGroup(EnvironmentDto environment) {
+        return null;
+    }
 
     String availabilityZone(NetworkRequest networkRequest, EnvironmentDto environment);
 
-    Set<String> getSubnets(NetworkRequest networkRequest);
+    default String availabilityZone(InstanceGroupNetworkRequest networkRequest, EnvironmentDto environment) {
+        return null;
+    }
+
+    Set<String> getSubnets(NetworkRequest networkRequest, boolean multiAzRequired);
```

Comment:
Again, this is certainly existing code, but why not drop `get` in the method name? It would be more consistent with the rest of the IF.